homeassistant/components/fritz/config_flow.py

```diff
@@ -0,0 +1,277 @@
+"""Config flow to configure the FRITZ!Box Tools integration."""
+import logging
+from urllib.parse import urlparse
+
+from fritzconnection.core.exceptions import FritzConnectionException
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.ssdp import (
+    ATTR_SSDP_LOCATION,
+    ATTR_UPNP_FRIENDLY_NAME,
+    ATTR_UPNP_UDN,
+)
+from homeassistant.config_entries import ConfigFlow
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+
+from .common import CONFIG_SCHEMA, FritzBoxTools
+from .const import DEFAULT_HOST, DEFAULT_PORT, DOMAIN, ERROR_CONNECTION_ERROR
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@config_entries.HANDLERS.register(DOMAIN)
+class FritzBoxToolsFlowHandler(ConfigFlow):
+    """Handle a FRITZ!Box Tools config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize FRITZ!Box Tools flow."""
+        self._host = None
+        self._entry = None
+        self._name = None
+        self._password = None
+        self._port = None
+        self._username = None
+        self.import_schema = None
+        self.fritz_tools = None
+
+    async def fritz_tools_init(self):
+        """Initialize FRITZ!Box Tools class."""
+        self.fritz_tools = FritzBoxTools(
+            hass=self.hass,
+            host=self._host,
+            port=self._port,
+            username=self._username,
+            password=self._password,
+        )
+
+        try:
+            await self.fritz_tools.async_setup()
+        except FritzConnectionException:
```

Comment:
FritzSecurityError should be handled here as well

Reply:
Done as well